src/dos/drive_local.cpp

```diff
@@ -46,32 +47,53 @@
 #include "cross.h"
 #include "inout.h"
 
-bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*/) {
-//TODO Maybe care for attributes but not likely
+bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t attributes) {
+	// Don't allow overwriting read-only files.
+	uint16_t test_attr = 0;
+	if (GetFileAttr(name, &test_attr) && (test_attr & DOS_ATTR_READ_ONLY)) {
+		DOS_SetError(DOSERR_ACCESS_DENIED);
+		return false;
+	}
+
+	bool existing_file = FileExists(name);
```

Comment:
`bool file_exists` reads better

Reply:
Going forward, we should use the nice `FatAttributeFlags` union that kcgen introduced. @weirddan455 has made good use of it in his MOVE command implementation, you can refer that for example usages.